File: backend/src/main/java/com/stockflow/nosql/domain/document/Product.java
package com.stockflow.nosql.domain.document;

import java.time.Instant;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "products")
public class Product {

    @Id
    private String id;
    private String name;

    @Indexed(unique = true)
    private String sku;

    private String categoryName;
    private double unitPrice;
    private boolean isActive;
    private int currentBalance;
    private String location;
    private Instant createdAtUtc;

    public Product() {
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSku() {
        return sku;
    }

    public void setSku(String sku) {
        this.sku = sku;
    }

    public String getCategoryName() {
        return categoryName;
    }

    public void setCategoryName(String categoryName) {
        this.categoryName = categoryName;
    }

    public double getUnitPrice() {
        return unitPrice;
    }

    public void setUnitPrice(double unitPrice) {
        this.unitPrice = unitPrice;
    }

    public boolean isActive() {
        return isActive;
    }

    public void setActive(boolean active) {
        isActive = active;
    }

    public int getCurrentBalance() {
        return currentBalance;
    }

    public void setCurrentBalance(int currentBalance) {
        this.currentBalance = currentBalance;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public Instant getCreatedAtUtc() {
        return createdAtUtc;
    }

    public void setCreatedAtUtc(Instant createdAtUtc) {
        this.createdAtUtc = createdAtUtc;
    }
}

File: mobile/stockflow_nosql/lib/core/models/product.dart
class Product {
  final String id;
  final String name;
  final String sku;
  final String categoryName;
  final double unitPrice;
  final bool isActive;
  final int currentBalance;
  final String? location;

  Product({
    required this.id,
    required this.name,
    required this.sku,
    required this.categoryName,
    required this.unitPrice,
    required this.isActive,
    required this.currentBalance,
    this.location,
  });

  factory Product.fromJson(Map<String, dynamic> json) => Product(
        id: json['id'] as String,
        name: json['name'] as String,
        sku: json['sku'] as String,
        categoryName: json['categoryName'] as String? ?? 'Unassigned category',
        unitPrice: (json['unitPrice'] as num).toDouble(),
        isActive: json['isActive'] as bool? ?? false,
        currentBalance: (json['currentBalance'] as num?)?.toInt() ?? 0,
        location: json['location'] as String?,
      );
}
